app/assets/javascripts/utilities/localTime.js

```diff
@@ -0,0 +1,13 @@
+/* Convert server string timestamp to local time, using the user's locale */
+function timestampToLocalTime(timestamp, options) {
+  if (timestamp === '') {
+    return '';
+  }
+
+  try {
+    var time = new Date(timestamp);
+    return new Intl.DateTimeFormat('default', options).format(time);
```

Comment:
To keep it simple, it could be `en-US` here and it'll be English for both displayed and on-hover. Although, it would be cool if it's displayed in English but `default` for on-hover but it's really optional.
```suggestion
    return new Intl.DateTimeFormat('en-US', options).format(time);
```

Reply:
@maestromac so we're not going to use the user's locale after all or do you mean we can do the switch (from `en-US` to `default`) later on when the article dates are ready to switch as well?

Reply:
Switching all to `default` later on is a good idea too, but that's not what I meant. Based on what I've tested, the first param for `DateTimeFormat` is just for displayed language. Time will still be formatted to user's local time.

Reply:
I put a browser in `it-IT` and then tested it:

```javascript
navigator.language
> "it-IT"
date.toLocaleString()
> "03/05/2019, 16:33:42"
new Intl.DateTimeFormat("default", { year: 'numeric', month: 'long', day: 'numeric', hour: 'numeric', minute: 'numeric', second: 'numeric' }).format(date)
> "3 May 2019, 16:33:42"
new Intl.DateTimeFormat("it-IT", { year: 'numeric', month: 'long', day: 'numeric', hour: 'numeric', minute: 'numeric', second: 'numeric' }).format(date)
> "3 maggio 2019, 16:33:42"
new Intl.DateTimeFormat("en-US", { year: 'numeric', month: 'long', day: 'numeric', hour: 'numeric', minute: 'numeric', second: 'numeric' }).format(date)
> "May 3, 2019, 4:33:42 PM"
```

As you can see, to have the full date string in English and the time formatted according to my own locale, then `default` is the correct one. If I force `en-US` then AM/PM are used. Which make sense.

If I switch my browser to `ja-JP` the same thing happens:

```javascript
navigator.language
> "ja"
new Intl.DateTimeFormat("default", { year: 'numeric', month: 'long', day: 'numeric', hour: 'numeric', minute: 'numeric', second: 'numeric' }).format(date)
> "3 May 2019, 16:41:29"
new Intl.DateTimeFormat("ja-JP", { year: 'numeric', month: 'long', day: 'numeric', hour: 'numeric', minute: 'numeric', second: 'numeric' }).format(date)
> "2019年5月3日 16:41:29"
new Intl.DateTimeFormat("en-US", { year: 'numeric', month: 'long', day: 'numeric', hour: 'numeric', minute: 'numeric', second: 'numeric' }).format(date)
> "May 3, 2019, 4:41:29 PM"
new Intl.DateTimeFormat(navigator.language, { year: 'numeric', month: 'long', day: 'numeric', hour: 'numeric', minute: 'numeric', second: 'numeric' }).format(date)
> "2019年5月3日 16:41:29"
navigator.languages
> (5) ["ja", "it-IT", "en-GB", "en-US", "en"]
```

This is what happens with the short date:

after setting the browser to `ja-JP`:

```javascript
new Intl.DateTimeFormat("en-US", { month: 'short', day: 'numeric' }).format(date)
> "May 3"
new Intl.DateTimeFormat("default", { month: 'short', day: 'numeric' }).format(date)
> "3 May"
```

after setting the browser to `it-IT`:

```javascript
new Intl.DateTimeFormat("default", { month: 'short', day: 'numeric' }).format(new Date())
> "3 May"
new Intl.DateTimeFormat("en-US", { month: 'short', day: 'numeric' }).format(new Date())
> "May 3"
```

It's a little confusing. If we want to keep `May 3` as the short date then we should force `en-US`, but if we want to have a localized hover then that should be use `navigator.language` because `default` is not reliable.

Let me know if that would work for you, I can change the functions to use two different locales. English for the short one, the user's locale (not `default` but `navigator.language`) with a fallback for the long one.

Fun moments with dates and times 😬

Reply:
That is confusing 😣

Your suggestion sounds great to me.

Reply:
I implemented the changes, this is how it looks:

Browser in English
![Screenshot 2019-05-03 at 6 38 52 PM](https://user-images.githubusercontent.com/146201/57152137-ee4b9000-6dd2-11e9-9d6c-473cdd0fdd80.png)

Browser in Italian
![Screenshot 2019-05-03 at 6 37 51 PM](https://user-images.githubusercontent.com/146201/57152144-f4417100-6dd2-11e9-89a9-3a962fbc6052.png)

Reply:
Awesome 👏